azure/deploy-to-azure/function_template.json

```diff
@@ -153,7 +153,7 @@
           typeof record.resourceId !== 'string'\r\n        ) {\r\n            return metadata;\r\n        }\r\n\r\n        var resourceId = this.createResourceIdArray(record);\r\n\r\n        if (resourceId[0] === 'subscriptions') {\r\n            if (resourceId.length > 1) {\r\n                metadata.tags.push('subscription_id:' + resourceId[1]);\r\n                if (resourceId.length == 2) {\r\n                    metadata.source = 'azure.subscription';\r\n                    return metadata;\r\n                }\r\n            }\r\n            if (resourceId.length > 3) {\r\n                if (\r\n                    resourceId[2] === 'providers' &&\r\n                    this.isSource(resourceId[3])\r\n                ) {\r\n                    \/\/ handle provider-only resource IDs\r\n                    metadata.source = this.formatSourceType(resourceId[3]);\r\n                } else {\r\n                    metadata.tags.push('resource_group:' + resourceId[3]);\r\n                    if (resourceId.length == 4) {\r\n                        metadata.source = 'azure.resourcegroup';\r\n                        return metadata;\r\n                    }\r\n                }\r\n            }\r\n            if (resourceId.length > 5 && this.isSource(resourceId[5])) {\r\n                metadata.source = this.formatSourceType(resourceId[5]);\r\n            }\r\n        } else if (resourceId[0] === 'tenants') {\r\n            if (resourceId.length > 3 && resourceId[3]) {\r\n                metadata.tags.push('tenant:' + resourceId[1]);\r\n                metadata.source = this.formatSourceType(resourceId[3]).replace(\r\n                    'aadiam',\r\n                    'activedirectory'\r\n                );\r\n            }\r\n        }\r\n        return metadata;\r\n    }\r\n}\r\n\r\nmodule.exports = async function(context, eventHubMessages) {\r\n    if (!DD_API_KEY || DD_API_KEY === '<DATADOG_API_KEY>') {\r\n        context.log.error(\r\n            'You must configure your API key before starting this function (see ## Parameters section)'\r\n        );\r\n        return;\r\n    }\r\n    try {\r\n        var handler = new EventhubLogHandler(context);\r\n        var parsedLogs = handler.handleLogs(eventHubMessages);\r\n    } catch (err) {\r\n        context.log.error('Error raised when parsing logs: ', err);\r\n        throw err;\r\n    }\r\n    var results = await new HTTPClient(context).sendAll(parsedLogs);\r\n\r\n    if (results.every(v => v === true) !== true) {\r\n        context.log.error(\r\n            'Some messages were unable to be sent. See other logs for details.'\r\n        );\r\n    }\r\n};\r\n\r\nmodule.exports.forTests = {\r\n    EventhubLogHandler,\r\n    Scrubber,\r\n    ScrubberRule,\r\n    Batcher,\r\n    constants: {\r\n        STRING,\r\n        STRING_ARRAY,\r\n        JSON_OBJECT,\r\n        JSON_ARRAY,\r\n        BUFFER_ARRAY,\r\n        JSON_STRING,\r\n        JSON_STRING_ARRAY,\r\n        INVALID\r\n    }\r\n};"
```

Comment:
Hard to tell in this diff but does this contain your new version with updated retries?

Reply:
Yes, it does. It's now version `0.5.7`. I also tested that the JSON escaping was correct by deploying a test log forwarder